pkg/core/system.go

```diff
@@ -235,28 +236,42 @@ func loadRelease(release string) ([]byte, error) {
 	return body, nil
 }
 
-func waitForIstioSidecarInjector(kc *kubectlClient) error {
-	print("Waiting for istio-sidecar-injector to start ")
+func waitForIstioComponents(kc *kubectlClient) error {
+	fmt.Print("Waiting for the Istio components to start ")
 	for i := 0; i < 36; i++ {
 		fmt.Print(".")
-		injectorStatus, err := kc.kubeCtl.Exec([]string{"get", "pod", "-n", istioNamespace, "-l", "istio=sidecar-injector", "-o", "jsonpath='{.items[0].status.phase}'"})
+		pods := kc.kubeClient.CoreV1().Pods(istioNamespace)
+		podList, err := pods.List(metav1.ListOptions{})
 		if err != nil {
-			// might take some time for the pod to show up so ignore early errors
-			if i > 3 {
-				return err
-			}
+			return err
 		}
-		if injectorStatus == "'Error'" {
-			return errors.New("istio-sidecar-injector pod failed to start")
+		waitLonger := false
+		for _, pod := range podList.Items {
+			if !strings.HasPrefix(pod.Name, "istio-") {
+				continue
+			}
+			if pod.Status.Phase != "Running" && pod.Status.Phase != "Succeeded" {
+				waitLonger = true
+				break
+			} else {
+				if pod.Status.Phase == "Running" {
+					containers := pod.Status.ContainerStatuses
+					for _, cont := range containers {
```

Comment:
unless the Pod is for a Job, in which case the container will not be ready, but completed successfully

Reply:
we check for 'Succeeded' which covers the job pods, see no reason to check the containers at that point. So anything 'Running' has to have all containers 'Ready', I guess a job pod could temporarily be in that state but I don't think that would cause issues for our purposes